### PR TITLE
[MAINTENANCE] Remove `include_rendered_content` flag

### PR DESCRIPTION
**Pull request by cdkini**
Rather than granular control over this, we should default to always rendering content for Cloud and omitting it otherwise.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/data_context/test_get_data_context.py [hunk 1]
```diff
@@ -272,24 +271,6 @@ def test_get_context_with_mode_equals_cloud_returns_cloud_data_context(
     assert isinstance(context, CloudDataContext)
 
 
-@pytest.mark.parametrize("ge_cloud_mode", [True, None])
-@pytest.mark.cloud
-def test_cloud_context_include_rendered_content(
```
Comment (billdirks) [inline tests/data_context/test_get_data_context.py:277]: Do we need some cloud test to ensure we are getting rendered content?

**Review by billdirks — COMMENTED** [review]
This lgtm. Is there a test that we include the rendered content when using cloud. It could be I missed it or is not in the diff since we already check for it and that check is unchanged.

File: tests/data_context/test_get_data_context.py (hunk 1 above)
Comment (cdkini) [reply tests/data_context/test_get_data_context.py:277]: This is just referring to the top level configuration (which doesn't exist anymore) - rendered content should exist on each object as attributes.

**Review by billdirks — APPROVED** [review]
Thank you!

**Pull request merged**